python-api/gstswitch/server.py

```diff
@@ -22,6 +26,110 @@
 TOOLS_DIR = '/'.join(os.getcwd().split('/')[:-1]) + '/tools/'
 
 
+class ProcessMonitor(subprocess.Popen):
+    """Runs a Command in a Background-Thread and monitors it's output
+
+    Can block until the Command prints a certain string and log the full
+    output into a file
+    """
+
+    def __init__(self, cmd, logtarget=sys.stderr):
+        self.log = logging.getLogger('server-output-monitor')
+
+        # Logfile to write to
+        self._logtarget = logtarget
+
+        # Internal Buffer to search for matched when wait_for_output is called
+        self._buffer = ""
+
+        self.log.debug("starting subprocess")
+
+        try:
+            super(ProcessMonitor, self).__init__(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                bufsize=0,
+                shell=False)
+        except Exception as err:
+            raise ServerProcessError(err)
+
+        self.log.debug("subprocess successfully started")
+
+    def terminate(self):
+        """Kills the process and waits for the thread to exit"""
+
+        self.log.info("reading remaining data from subprocess")
+        while True:
+            (read, _, _) = select.select([self.stdout], [], [], 0)
+            if self.stdout not in read:
+                break
+
+            self.log.debug("reading data from subprocess")
+            chunk = os.read(self.stdout.fileno(), 2000).decode('utf-8')
```

Comment:
Can you add a comment about why you are using `os.read` here. That method is super uncommon and every time I see this, I think "that is clearly wrong" and then remember our previous discussions around it.